bcipy/helpers/stimuli.py

```diff
@@ -227,6 +228,144 @@ def extract_trials(
         return np.stack(new_trials, 1)  # C x T x S
 
 
+class GazeReshaper:
+    def __call__(self,
+                 trial_targetness_label: List[str],
+                 inq_start_times: List[float],
+                 target_symbols: List[str],
+                 gaze_data: np.ndarray,
+                 sample_rate: int,
+                 trials_per_inquiry: int,
+                 offset: float = 0,
+                 channel_map: List[int] = None,
+                 poststimulus_length: float = 0.5,
+                 prestimulus_length: float = 0.0,
+                 transformation_buffer: float = 0.0,
+                 ) -> dict:
+        """Extract inquiry data and labels.
+
+        Args:
+            trial_targetness_label (List[str]): labels each trial as "target", "non-target", "first_pres_target", etc
+            timing_info (List[float]): Timestamp of each event in seconds
+            gaze_data (np.ndarray): shape (channels, samples) eye tracking data
+            sample_rate (int): sample rate of data provided in eeg_data
+            trials_per_inquiry (int): number of trials in each inquiry
+            offset (float, optional): Any calculated or hypothesized offsets in timings. Defaults to 0.
+            channel_map (List[int], optional): Describes which channels to include or discard.
+                Defaults to None; all channels will be used.
+            poststimulus_length (float, optional): time in seconds needed after the last trial in an inquiry.
+            prestimulus_length (float, optional): time in seconds needed before the first trial in an inquiry.
+            transformation_buffer (float, optional): time in seconds to buffer the end of the inquiry. Defaults to 0.0.
+
+        Returns:
```

Comment:
It looks like it is returning the data_by_targets only

Reply:
I added more documentation to the reshaper, updated some return values as `optional` and added their use cases in the docstring.